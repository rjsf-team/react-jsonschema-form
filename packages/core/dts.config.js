// Not transpiled with TypeScript or Babel, so use plain Es6/Node.js!
/**
 * @type {import('dts-cli').DtsConfig}
 */
module.exports = {
  // This function will run for each entry/format/env combination
  rollup(config, options) {
    config.output.name = 'JSONSchemaForm';
    return config; // always return a config.
  },
};
