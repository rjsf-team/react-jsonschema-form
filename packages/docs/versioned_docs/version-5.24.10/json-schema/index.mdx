---
title: JSON Schema
description: Documentation for how to use JSON Schema constructs in react-jsonschema-form.,
---

import DocCardList from '@theme/DocCardList';

<p>{frontMatter.description}</p>

<DocCardList  />