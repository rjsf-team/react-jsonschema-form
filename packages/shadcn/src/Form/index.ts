export { default } from './Form';
export * from './Form';
