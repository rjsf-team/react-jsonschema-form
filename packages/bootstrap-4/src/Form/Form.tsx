import { withTheme, FormProps } from "@rjsf/core";

import Theme from "../Theme";
import { StatelessComponent } from "react";

const Form:
  | React.ComponentClass<FormProps<any>>
  | StatelessComponent<FormProps<any>> = withTheme(Theme);

export default Form;
