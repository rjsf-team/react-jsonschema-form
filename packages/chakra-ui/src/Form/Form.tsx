import { withTheme, FormProps } from "@rjsf/core";

import Theme from "../Theme";
import { ComponentClass, FunctionComponent } from "react";

const Form:
  | ComponentClass<FormProps<any>>
  | FunctionComponent<FormProps<any>> = withTheme(Theme);

export default Form;
