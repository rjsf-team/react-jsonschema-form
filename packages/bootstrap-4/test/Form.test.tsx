import React from "react";
import Form from "../src/index";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";
import { UiSchema } from "@rjsf/core";

describe("single fields", () => {
  describe("string field", () => {
    test("regular", () => {
      const schema: JSONSchema7 = {
        type: "string",
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("format email", () => {
      const schema: JSONSchema7 = {
        type: "string",
        format: "email",
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("format uri", () => {
      const schema: JSONSchema7 = {
        type: "string",
        format: "uri",
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("format data-url", () => {
      const schema: JSONSchema7 = {
        type: "string",
        format: "data-url",
      };
      const tree = renderer.create(<Form schema={schema} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  test("string field with placeholder", () => {
    const schema: JSONSchema7 = {
      type: "string",
    };
    const uiSchema: UiSchema = {
      "ui:placeholder": "placeholder",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("number field", () => {
    const schema: JSONSchema7 = {
      type: "number",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("number field 0", () => {
    const schema: JSONSchema7 = {
      type: "number",
    };
    const formData = 0;
    const tree = renderer
      .create(<Form schema={schema} formData={formData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("null field", () => {
    const schema: JSONSchema7 = {
      type: "null",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("unsupported field", () => {
    const schema: JSONSchema7 = {
      type: undefined,
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("format color", () => {
    const schema: JSONSchema7 = {
      type: "string",
      format: "color",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("format date", () => {
    const schema: JSONSchema7 = {
      type: "string",
      format: "date",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("format datetime", () => {
    const schema: JSONSchema7 = {
      type: "string",
      format: "datetime",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("password field", () => {
    const schema: JSONSchema7 = {
      type: "string",
    };
    const uiSchema: UiSchema = {
      "ui:widget": "password",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("textarea field", () => {
    const schema: JSONSchema7 = {
      type: "string",
    };
    const uiSchema: UiSchema = {
      "ui:widget": "textarea",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("select field", () => {
    const schema: JSONSchema7 = {
      type: "string",
      enum: ["foo", "bar"],
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("checkbox field", () => {
    const schema: JSONSchema7 = {
      type: "boolean",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("checkbox field", () => {
    const schema: JSONSchema7 = {
      type: "boolean",
    };
    const tree = renderer.create(<Form schema={schema} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("checkboxes field", () => {
    const schema: JSONSchema7 = {
      type: "array",
      items: {
        type: "string",
        enum: ["foo", "bar", "fuzz", "qux"],
      },
      uniqueItems: true,
    };
    const uiSchema: UiSchema = {
      "ui:widget": "checkboxes",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("radio field", () => {
    const schema: JSONSchema7 = {
      type: "boolean",
    };
    const uiSchema: UiSchema = {
      "ui:widget": "radio",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  // There is a bug in the Material UI <Slider /> component that prevents this from working.
  // Error: `TypeError: Cannot read property 'addEventListener' of null`
  // From: https://github.com/mui-org/material-ui/blob/v4.5.2/packages/material-ui/src/Slider/Slider.js#L622
  test.skip("slider field", () => {
    const schema: JSONSchema7 = {
      type: "integer",
      minimum: 42,
      maximum: 100,
    };
    const uiSchema: UiSchema = {
      "ui:widget": "range",
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("hidden field", () => {
    const schema: JSONSchema7 = {
      type: "object",
      properties: {
        "my-field": {
          type: "string",
        },
      },
    };
    const uiSchema: UiSchema = {
      "my-field": {
        "ui:widget": "hidden",
      },
    };
    const tree = renderer
      .create(<Form schema={schema} uiSchema={uiSchema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
