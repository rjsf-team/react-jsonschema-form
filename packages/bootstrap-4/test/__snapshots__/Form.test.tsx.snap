// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single fields checkbox field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="checkbox  form-group"
    >
      <div
        className="form-check"
      >
        <input
          autoFocus={false}
          checked={false}
          className="form-check-input position-static"
          disabled={false}
          id="root"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields checkbox field 2`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="checkbox  form-group"
    >
      <div
        className="form-check"
      >
        <input
          autoFocus={false}
          checked={false}
          className="form-check-input position-static"
          disabled={false}
          id="root"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields checkboxes field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="root"
    />
    <div
      className="form-group"
    >
      <form
        className=""
      >
        <div
          className="bg-transparent border-0 custom-control custom-checkbox"
        >
          <input
            autoFocus={false}
            checked={false}
            className="custom-control-input"
            disabled={false}
            id="root_0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="root_0"
            title=""
          >
            foo
          </label>
        </div>
      </form>
      <form
        className=""
      >
        <div
          className="bg-transparent border-0 custom-control custom-checkbox"
        >
          <input
            autoFocus={false}
            checked={false}
            className="custom-control-input"
            disabled={false}
            id="root_1"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="root_1"
            title=""
          >
            bar
          </label>
        </div>
      </form>
      <form
        className=""
      >
        <div
          className="bg-transparent border-0 custom-control custom-checkbox"
        >
          <input
            autoFocus={false}
            checked={false}
            className="custom-control-input"
            disabled={false}
            id="root_2"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="root_2"
            title=""
          >
            fuzz
          </label>
        </div>
      </form>
      <form
        className=""
      >
        <div
          className="bg-transparent border-0 custom-control custom-checkbox"
        >
          <input
            autoFocus={false}
            checked={false}
            className="custom-control-input"
            disabled={false}
            id="root_3"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="root_3"
            title=""
          >
            qux
          </label>
        </div>
      </form>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields format color 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="color"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields format date 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="date"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields format datetime 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="datetime-local"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields hidden field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="p-0 container-fluid"
    >
      <div
        className="d-none row"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <div
          className="col-12"
        >
           
          <div
            className="form-group"
          >
            <input
              id="root_my-field"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields null field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields number field 0 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="number"
        value={0}
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields number field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="number"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields password field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        type="password"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields radio field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="d-block form-label"
      />
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          disabled={false}
          id="Yes"
          name="root"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value={true}
        />
        <label
          className="form-check-label"
          htmlFor="Yes"
          title=""
        >
          Yes
        </label>
      </div>
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          disabled={false}
          id="No"
          name="root"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value={false}
        />
        <label
          className="form-check-label"
          htmlFor="No"
          title=""
        >
          No
        </label>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields select field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      />
      <select
        autoFocus={false}
        className="custom-select"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        value=""
      >
        <option
          value=""
        >
          
        </option>
        <option
          disabled={false}
          id="foo"
          value="foo"
        >
          foo
        </option>
        <option
          disabled={false}
          id="bar"
          value="bar"
        >
          bar
        </option>
      </select>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields string field format data-url 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control-file"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="file"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields string field format email 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="email"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields string field format uri 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="url"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields string field regular 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields string field with placeholder 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="mb-0 form-group"
    >
      <label
        className="form-label"
      />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="placeholder"
        readOnly={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields textarea field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="root"
    />
    <div
      className="input-group"
    >
      <textarea
        autoFocus={false}
        className="form-control"
        disabled={false}
        id="root"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        readOnly={false}
        rows={5}
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`single fields unsupported field 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="unsupported-field"
    >
      <p>
        Unsupported field schema
        <span>
           for
           field 
          <code>
            root
          </code>
        </span>
        <em>
          : 
          Unknown field type undefined
        </em>
        .
      </p>
      <pre>
        {}
      </pre>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
